address-controller/src/main/java/io/enmasse/controller/common/KubernetesHelper.java

```diff
@@ -344,6 +352,82 @@ public boolean isDestinationClusterReady(String clusterId) {
         return Optional.ofNullable(client.secrets().inNamespace(namespace).withName(secretName).get());
     }
 
+    private JsonObject doRawHttpRequest(String path, JsonObject body) {
+        OkHttpClient httpClient = client.adapt(OkHttpClient.class);
+
+        HttpUrl url = HttpUrl.get(client.getOpenshiftUrl()).resolve(path);
+        Request request = new Request.Builder()
+                .url(url)
+                .addHeader("Content-Type", "application/json")
+                .addHeader("Authorization", "Bearer " + controllerToken)
+                .post(RequestBody.create(MediaType.parse("application/json"), body.encode()))
+                .build();
+        try {
+            Response response = httpClient.newCall(request).execute();
+            if (response.isSuccessful()) {
+                return new JsonObject(response.body().string());
+            } else {
+                String errorMessage = String.format("Error performing POST on %s: %d, %s", path, response.code(), response.body());
+                log.warn(errorMessage);
+                throw new RuntimeException(errorMessage);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+
+    @Override
+    public TokenReview performTokenReview(String token) {
+        if (client.isAdaptable(OkHttpClient.class)) {
+            JsonObject body = new JsonObject();
+
+            body.put("kind", "TokenReview");
+            body.put("apiVersion", "authentication.k8s.io/v1beta1");
+
+            JsonObject spec = new JsonObject();
+            spec.put("token", token);
+            body.put("spec", spec);
+
+            JsonObject responseBody= doRawHttpRequest("/apis/authentication.k8s.io/v1beta1/tokenreviews", body);
+            JsonObject status = responseBody.getJsonObject("status");
+            boolean authenticated = status.getBoolean("authenticated");
+            String userName = null;
+            JsonObject user = status.getJsonObject("user");
+            if (user != null) {
+                userName = user.getString("username");
+            }
+            return new TokenReview(userName, authenticated);
+        } else {
+            return new TokenReview(null, false);
+        }
+    }
+
+    @Override
+    public SubjectAccessReview performSubjectAccessReview(String user, String path, String verb) {
+        if (client.isAdaptable(OkHttpClient.class)) {
```

Comment:
How come you're not using OpenShiftClient.subjectAccessReviews() instead? Is it because you want to support Kubernetes instead of just OpenShift?

Reply:
Yes, I was thinking I had already crossed a 'limit' when using the http client directly, so might as well use the same API for these things at least.